Source/WTF/wtf/ThreadSafeWeakPtr.h

```diff
@@ -142,7 +142,11 @@ class ThreadSafeRefCountedAndCanMakeThreadSafeWeakPtr {
 private:
     template<typename> friend class ThreadSafeWeakPtr;
     template<typename> friend class ThreadSafeWeakHashSet;
-    ThreadSafeWeakPtrControlBlock& m_controlBlock { *new ThreadSafeWeakPtrControlBlock(static_cast<T&>(*this)) };
+#if COMPILER(MSVC)
```

Comment:
It would be nice if we could use the same code with all compilers.  I'm aware of this issue with MSVC.  Could we just use the MSVC code everywhere?

Reply:
We probably could use a C-style cast everywhere, but I'd rather have the type safety that comes with `static_cast<>` on clang/gcc and just workaround MSVC.

I could also come up with a terrible macro to combine the C-style cast and the `static_cast<>` if you'd prefer that.